ee/control/client_http.go

```diff
@@ -70,17 +72,23 @@ func (c *HTTPClient) GetConfig(ctx context.Context) (io.Reader, error) {
 	ctx, span := traces.StartSpan(ctx)
 	defer span.End()
 
-	challengeReq, err := http.NewRequestWithContext(ctx, http.MethodGet, c.url("/api/agent/config").String(), nil)
+	challengeCtx, challengeCancel := context.WithTimeout(ctx, defaultRequestTimeout)
```

Comment:
this works, but couldn't we simplify just by putting the logic in the `do()` func?

something like
```go

	ctx, cancel := context.WithTimeout(req.Context(), defaultRequestTimeout)
	defer cancel()

	req = req.WithContext(ctx)
```

Reply:
I wasn't sure if we might want different timeouts in the future for different endpoints

Reply:
fair enough

Reply:
Decided we probably didn't want different timeouts, so I updated to move it to `do()`!

Missed your other comment -- I think we might've only been passing that ctx in for the traces anyway. 😅 I'll update!

Reply:
Okay, fixed both things! LMK how it looks